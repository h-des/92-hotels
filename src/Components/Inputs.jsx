import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Input = styled.input`
  font-size: 1.4rem;
  font-weight: 600;
  font-family: 'Nunito', sans-serif;
  color: ${props => props.theme.colors.black};
  padding: 0.9rem 2rem;
  border: 2px solid transparent;
  width: 100%;
  min-height: 40px;
  outline: none;
  border-radius: 3px;
  background-color: #f4f5f7;
  margin-bottom: ${props => (props.marginBottom ? props.marginBottom : `0`)};

  &:focus {
    background-color: transparent;
    border: 2px solid #4c9aff;
  }
`;

const StyledDiv = styled.div`
  display: flex;
  flex: 1 1 0;
  min-height: 40px;
  margin-bottom: ${props => (props.marginBottom ? props.marginBottom : `0`)};
`;

const StyledButton = styled.button`
  width: 20px;
  height: 40px;
  margin: 0 3px;
  background-color: transparent;
  color: ${props => props.theme.colors.primary};
  font-weight: 700;
  border: none;
  font-size: 18px;
  cursor: pointer;

  &:hover {
    color: ${props => props.theme.colorsHover.primary};
  }
`;

const NumberInput = ({
  currentValue,
  onPlusClick,
  onMinusClick,
  fieldName,
  onChangeEvent,
  marginBottom
}) => {
  return (
    <StyledDiv marginBottom={marginBottom}>
      <StyledButton onClick={e => onMinusClick(e, fieldName)}>-</StyledButton>
      <Input
        name={fieldName}
        id={fieldName}
        min={0}
        onChange={onChangeEvent}
        value={currentValue}
      />
      <StyledButton onClick={e => onPlusClick(e, fieldName)}>+</StyledButton>
    </StyledDiv>
  );
};

NumberInput.propTypes = {
  onChangeEvent: PropTypes.func,
  onPlusClick: PropTypes.func,
  onMinusClick: PropTypes.func,
  currentValue: PropTypes.number,
  fieldName: PropTypes.string
};

export { Input, NumberInput };
