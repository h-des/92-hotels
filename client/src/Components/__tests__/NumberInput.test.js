import React from 'react';
import { shallow } from 'enzyme';
import { NumberInput } from '../Inputs';

describe('Test NumberInput', () => {
  let wrapper;
  const mockOnClickPlus = jest.fn();
  const mockOnClickMinus = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <NumberInput
        fieldName="test"
        onMinusClick={mockOnClickMinus}
        onPlusClick={mockOnClickPlus}
        currentValue={10}
      />
    );
  });

  test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Set input name', () => {
    expect(wrapper.find("[name='test']")).toHaveLength(1);
  });

  test('Test minus click button', () => {
    wrapper.find("[children='-']").simulate('click');
    expect(mockOnClickMinus.mock.calls.length).toEqual(1);
  });

  test('Test plus click button', () => {
    wrapper.find("[children='+']").simulate('click');
    expect(mockOnClickPlus.mock.calls.length).toEqual(1);
  });

  test('Set current val', () => {
    expect(wrapper.find('[value=10]')).toHaveLength(1);

    wrapper.setProps({ currentValue: 2 });
    expect(wrapper.find('[value=2]')).toHaveLength(1);
  });
});
