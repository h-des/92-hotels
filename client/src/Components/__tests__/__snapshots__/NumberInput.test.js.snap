// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NumberInput Renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NumberInput
    currentValue={10}
    fieldName="test"
    onMinusClick={[MockFunction]}
    onPlusClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.button
          onClick={[Function]}
        >
          -
        </styled.button>,
        <styled.input
          id="test"
          min={0}
          name="test"
          value={10}
        />,
        <styled.button
          onClick={[Function]}
        >
          +
        </styled.button>,
      ],
      "marginBottom": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "-",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "-",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "test",
          "min": 0,
          "name": "test",
          "onChange": undefined,
          "value": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "+",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.button
            onClick={[Function]}
          >
            -
          </styled.button>,
          <styled.input
            id="test"
            min={0}
            name="test"
            value={10}
          />,
          <styled.button
            onClick={[Function]}
          >
            +
          </styled.button>,
        ],
        "marginBottom": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "-",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "-",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "test",
            "min": 0,
            "name": "test",
            "onChange": undefined,
            "value": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "+",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
